x-pack/plugins/security/server/authentication/index.ts

```diff
@@ -105,28 +105,25 @@ export async function setupAuthentication({
       // authentication (username and password) or arbitrary external page managed by 3rd party
       // Identity Provider for SSO authentication mechanisms. Authentication provider is the one who
       // decides what location user should be redirected to.
-      return t.redirected(authenticationResult.redirectURL!);
+      return response.redirected(undefined, {
```

Comment:
__note:__ feels a bit weird to have a dedicated `redirected` method, but leverage _optional_ generic `headers` for the _required_ `location` header.... But functionally it works for us, so :man_shrugging:

Reply:
I kinda agree. However, we did it to be able to send a payload with redirection response and to comply with the other methods signature `(payload, options) => KibanaResponse`